emily/handler/src/database/entries/mod.rs

```diff
@@ -1,8 +1,435 @@
+                        .build()?,
+                )
+                .build();
+            write_delete_requests.push(write_request);
+        }
+        // Execute the deletes in chunks.
+        for chunk in write_delete_requests.chunks(25) {
+            dynamodb_client
+                .batch_write_item()
+                .request_items(table_name, chunk.to_vec())
+                .send()
+                .await?;
+        }
+        // Return.
+        Ok(())
+    }
+
+    /// Wipes the table.
+    #[cfg(feature = "testing")]
+    async fn wipe(
+        dynamodb_client: &aws_sdk_dynamodb::Client,
+        settings: &Settings,
+    ) -> Result<(), Error> {
+        // Get table name.
+        match Self::INDEX_NAME_IF_GSI {
+            // If this is secondary index, so go to primary:
+            Some(_) => {
+                <Self::PrimaryIndex as TableIndexTrait>::delete_entries(
+                    dynamodb_client,
+                    settings,
+                    <Self::PrimaryIndex as TableIndexTrait>::get_all_entries(
+                        dynamodb_client,
+                        settings,
+                    )
+                    .await?
+                    .iter()
+                    .map(<<Self::PrimaryIndex as TableIndexTrait>::Entry as EntryTrait>::key)
+                    .collect(),
+                )
+                .await
+            }
+            // Otherwise this is a primary index so wipe the table.
+            None => {
+                Self::delete_entries(
+                    dynamodb_client,
+                    settings,
+                    Self::get_all_entries(dynamodb_client, settings)
+                        .await?
+                        .iter()
+                        .map(<Self::Entry as EntryTrait>::key)
+                        .collect(),
+                )
+                .await
+            }
+        }
+    }
+}
+
+/// Secondary index search token definition.
+#[derive(Clone, Default, Debug, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub struct SecondaryIndexSearchToken<P, S> {
+    /// Primary index.
+    #[serde(flatten)]
+    primary: P,
+    /// Secondary index.
+    #[serde(flatten)]
+    secondary: S,
+}
+
+// Private Helpers -------------------------------------------------------------
+
+/// Converts an optional `HashMap<String, AttributeValue>` representing the last evaluated key
+/// into an optional token string. If the `Option` contains a value, it is deserialized into a type `T`
+/// and then serialized into a token string.
+fn maybe_next_token_from_last_evaluated_key<T>(
+    maybe_last_evaluated_key: Option<HashMap<String, AttributeValue>>,
+) -> Result<Option<String>, Error>
+where
+    T: Serialize + for<'de> Deserialize<'de>,
+{
+    let maybe_next_token = maybe_last_evaluated_key
+        .map(|key| serde_dynamo::from_item(key))
+        .transpose()?
+        .map(|token_data: T| tokenize(token_data))
```

Comment:
I was going to ask: why map this to a string? But then I saw that we return these `last_evaluated_key` tokens to the client as a string, so this all makes sense.